blog/tidy-market-microstructure/index.qmd

```diff
@@ -0,0 +1,1674 @@
+
+```{r}
+#| echo=FALSE
+library(kableExtra)
+
+variable_table <- data.frame(
+  "Variable name" = c("Quoted bid-ask spread", "Quoted depth", "Effective bid-ask spread", "Trade volume", "Price impact", "Realized spread", "Return autocorrelation", "Realized volatility", "Variance ratio"),
+  "Variable type" = c("Liquidity", "Liquidity", "Liquidity", "Volume", "Liquidity", "Liquidity", "Efficiency", "Volatility", "Efficiency"),
+  "Data type" = c("Quote tick data", "Quote tick data", "Quote and trade tick data", "Trade tick data", "Quote and trade tick data", "Quote and trade tick data", "Equispaced quote data", "Equispaced quote data", "Equispaced quote data"),
+  "Section" = c(1, 1, 2, 2, 2, 2, 3, 3, 3),
+  check.names = FALSE
+  )
+
+kable(variable_table,
+      caption = "Market quality variables and data types",
+      booktabs = TRUE,
+      longtable = FALSE)
+```
+
+Our focus is on the *practice* of empirical market microstructure. As such, we often motivate the coding choices with economic concepts. Other than that, however, readers interested in the economics underlying each metric are referred to introductory texts such as Campbell, Lo and MacKinlay (1997)[^4], Foucault, Pagano & Röell (2013)[^5] and Hasbrouck (2007)[^6]. 
+
+**Programming.**
+The coding here is in R.
+
+The choice of language is of course a matter of taste. When deciding on the tools to use, what is important to consider is that the data files in this field are often huge. How can we process them without crashing our laptops? The example data used in this guide are tiny, but we wrote the code to be efficient when the number of observations per day go from thousands to millions.^[A previous guide to microstructure programming (in SAS) is provided by Boehmer, Broussard and Kallunki (2002): Boehmer, E., Broussard, J. P., & Kallunki, J. P. (2002). *Using SAS in financial research*. SAS Publishing.]
+
+Even within the world of R, there are of course different packages available to pursue the same goal. We primarily rely on `data.table`, which is a very fast package for working with the high dimensional data that characterizes microstructure. The speed benefits show when reading and sorting data, and really shine when applying functions to several groups.^[Part of the `data.table` speed advantage is that it uses parallel processing by default. Consequently, one has to be cautious to use it in parallelization, as improper specification may lead to [over-parallelization](https://towardsdatascience.com/parallelization-caveats-in-r-1-the-basics-multiprocessing-and-multithreading-performance-eb584b7e850e) and undermine the performance benefits.] 
```

Comment:
Please add something like "Since the example data in this post is fairly small with around 250k rows, there are hardly any performance benefits of data.table over other packages. In practice, however, one works with billions of rows and the benefits of data.table manifest quickly."